homeassistant/components/alarmdecoder/entity.py

```diff
@@ -0,0 +1,18 @@
+"""Support for AlarmDecoder-based alarm control panels entity."""
+from homeassistant.helpers.entity import Entity
+
+from .const import (
+    DOMAIN,
+)
+
+class AlarmDecoderEntity(Entity):
+    """Define a base AlarmDecoder entity."""
+
+    @property
+    def device_info(self):
+        return {
+            "identifiers": {(DOMAIN, self._client.serial_number)},
+            "manufacturer": "NuTech",
+            "serial_number": self._client.serial_number,
+            "sw_version": self._client.version_number,
+        }
```

Comment:
```suggestion
class AlarmDecoderEntity(Entity):
    """Define a base AlarmDecoder entity."""
    
    ```suggestion
class AlarmDecoderEntity(Entity):
    """Define a base AlarmDecoder entity."""

    _attr_has_entity_name = True
    
    def __init__(self, client):
        self._client = client
        self._attr_device_info = DeviceInfo(
            identifiers={(DOMAIN, self._client.serial_number)},
            manufacturer="NuTech",
            serial_number=self._client.serial_number,
            sw_version=self._client.version_number,
        )
```

Reply:
I think you can also set `entry_type` on the device_info